### PR TITLE
Improve the Basic Authentication section

### PR DESCRIPTION
**Pull request by MichalMaler**
Improve the Basic Authentication section

Adding intro for the BA section and also the information about how to configure it.

Close:
https://issues.redhat.com/browse/QDOCS-27
https://issues.redhat.com/browse/QDOCS-28

### CODE REVIEW

**Comment by gsmet** [comment]
Hi @MichalMaler ,

A few comments on the form of the pull request:
- Please make the title of the PR a sentence similar to `Improve the Basic Authentication section` - it's important as they are included in our community release notes
- And then include the issue numbers in the description as you did
- And same for the commit comment

If you form your commit comment like:
```
Improve the Basic Authentication section

Close:
https://issues.redhat.com/browse/QDOCS-27
https://issues.redhat.com/browse/QDOCS-28
```
(note the blank line, it's important)

Then GitHub will automatically fill the fields appropriately when creating the pull request if you have only one commit and your work will be easier.

/cc @ebullient @quarkusio/documentation for general awareness.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:19]: We never used this construct without an admonition type (e.g. `[NOTE]`) so we need to check how it renders on the website.

/cc @ebullient

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:30]: Same here, we usually use plain title levels, I have no idea how it renders on the website.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:38]: For source code, we enforce:
`[source,<type>]` so in this case `[source,properties]`

**Comment by MichalMaler** [comment]
@gsmet Hello there. Thank you for the feedback. I applied what I could. Let see how it looks on the portal and if Erin will like it.
BTW, I tried to used the `./mvnw -f docs clean install` command to build it locally, but I run into some 999-SNAPSHOT error.
Any idea how to make it work, please?

Cheers!

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:19]: @gsmet Do you mean this one, `_Authorization: Basic_ _<credentials>_` ?
I'd just go for a simple highlighting like `Authorization: Basic <credentials>`

Comment (MichalMaler) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:19]: @sberyozkin I think he was talking about the ==== box I used to highlight the example.
The underscores mean a plain italic.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:10]: The change from `and Mutual TLS authentication as well as the ` to `and both Mutual TLS and proactive authentication` is not quite correct, `proactive authentication` is not a built in mechanism but is about controlling at what point these built in and other authentication mechanisms start working. So I think it needs to be reverted. 
Perhaps we'll move the proactive authentication section to its own doc eventually

Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:10]:
```suggestion
The following section describes the Quarkus built-in authentication mechanisms for HTTP-based BASIC, FORM and Mutual TLS authentication. Proactive authentication is also described.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:10]: May be like this ?

Comment (MichalMaler) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:10]: Ok, putting it back and removing that proactive mention.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:15]: I'm not sure having a `BA` acronym is useful, I'd rather remove it

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:18]: Yes, what @MichalMaler said.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:14]: Yeah, tbh, I had the same comment and refrained from doing it. I have never seen it in the wild (I just saw it on the Wikipedia page while checking if it was a thing).

**Comment by gsmet** [comment]
@MichalMaler yeah, you need to do a full build first. Do a `./mvnw -DquicklyDocs` then wait for a long time and you'll be all set.

Then you can iterate on the docs module as you wanted to do it without doing a full build every time. You might have to do it from time to time when we add a new module but it doesn't happen that often these days so you should be pretty much OK.

**Review by hmanwani-rh — COMMENTED** [review]
@MichalMaler Thank you for creating this PR. The changes look great. 
I'm adding a few suggestions to the changes to align with the style guide, please take a look and let me know if you've any questions.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:17]:
```suggestion
In the context of an HTTP request, Basic authentication is a method for an HTTP user agent (such as a Web browser) to provide a user name and password when making a request. A request contains a header field in the form of _Authorization: Basic_ _<credentials>_, where credentials are the Base64 encoding of a user ID and password joined by a colon. 
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:17]: `Basic authentication` is used everywhere so I propose to use the same phrase. Also, if you don't mind, please change `Basic authentication` to `Basic Authentication`, it seems like the latter option is more frequently used (even in the PR title :-) ) - but I'll leave it up to you

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:22]:
```suggestion
If the user name is `Alice` and the password is `secret`, the HTTP Authorization header will look as `Authorization: Basic  QWxjZTpzZWNyZXQ=`, where `QWxjZTpzZWNyZXQ=` is a Base64 encoded representation of the `Alice:secret` string.
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:36]: Star is OK here, since we can use it with the procedures with only one step.

File: docs/src/main/asciidoc/security.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security.adoc:208]: I agree, but I need to be consistent with the whole paragraph, the rewriting of which is out of scope for this issue.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc]: IMHO the original proposal is correct. Though a `username/password based IdentityProvider` may indeed cause some confusion. So how about `an IdentityProvider which uses a username and password to create a `SecurityIdentity`` ?

File: docs/src/main/asciidoc/security.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security.adoc:216]:
```suggestion
You can also use a xref:security-testing.adoc#configuring-user-information[User Properties IdentityProvider] to enable Basic Authentication however it is recommended to use this 'IdentityProvider' for testing only because the user names, passwords and roles are set directly in `application.properties`.
```

File: docs/src/main/asciidoc/security.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security.adoc:213]:
```suggestion
* For a a Basic Authentication configuration walk-through using JPA, see:
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:15]:
```suggestion
HTTP Basic Authentication uses fields in the HTTP header and is the easiest to set up. Also, it is one of the least resource-demanding techniques that provides user credentials to the secured Web resources without requiring HTTP cookies, session identifiers, or login pages.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:15]: Minor suggestion, since it is a secured Web Resource which enforces the access control, not the actual authentication technique

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:17]:
```suggestion
In the context of an HTTP request, Basic Authentication is a method for an HTTP user agent, such as a web browser, to provide a user name and password when creating a request. In Basic Authentication, a request contains a header field in the form of `Authorization: Basic <credentials>`, where credentials are the Base64 encoding of a user ID and password joined by a colon as described in the following example.
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:43]:
```suggestion
For a Basic Authentication configuration walk-through that uses `JPA` see the xref:security-getting-started.adoc[Getting Started With Security] guide.
```

**Comment by sberyozkin** [comment]
@MichalMaler @hmanwani-rh Thanks very much for this effort, LGTM, sorry if some of my suggestions look a bit pedantic. Your PR improves Basic Authentication related documentation a lot. I guess it is only the first step, eventually it will be moved into the dedicated doc(s)

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (ebullient) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:19]: `Authorization: Basic <credentials>` should be inline code, rather than italics, I believe. 

Without the type (e.g. NOTE), it is a [simple admonition](https://docs.asciidoctor.org/asciidoc/latest/blocks/example-blocks/#delimited) that should render as a plain callout box, but we will have to see what the web CSS does with it.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (ebullient) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:36]: Again, we use less formal language. Product docs use 'Procedure' traditionally, but that's a less friendly/open tone.

If there is only one step, I think you can skip the bullet entirely.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (ebullient) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:34]:
```suggestion
```

Community docs do not usually use the word Procedure, as it is a more formal tone than we usually want. Given this has a single step, and is immediately below a heading, I think it can be a simple paragraph

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (ebullient) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:29]: Yes. I don't think the intro sentence is necessary

File: docs/src/main/asciidoc/security-built-in-authentication.adoc
Comment (ebullient) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:10]:
```suggestion
Explore Quarkus built-in authentication mechanisms and behaviors. Quarkus supports HTTP FORM, BASIC, and Mutual TLS authentication and will proactively authenticate users by default.
```

Do we need words like "the following section" in here at all? In most cases, that scope is clearly implied, making it redundant. This should be a hook: why do I want to read this? The typical (in my past) word for that was "Learn.." but diataxis has a good justification for why that isn't the best choice.

Abstracts are hard, but I would like to use these for the tiles on the website at some point (automagically to reduce manual fiddling), which requires being concise (another reason to remove the "In this document.. " phrasing).

**Review by ebullient — COMMENTED** [review]
Just to confirm, this is a first pass at revising an existing doc, right? (I think I remember this).. we're kind of in the fuzzy middle to me on this one (a cross between a how-to and an explanation... )

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
LGTM, I can merge shortly, my understanding is that right now the task is about getting the content right, the actual restructuring to align with new templates will follow up

**Comment by sberyozkin** [comment]
@ebullient right, just saw your comment, yes, that is my understanding too. The fist pass is about revising the current content for different security documents, preparing them really for the next phase and then in the next phase it will be a matter of moving the chunks of content around to align with the new model

**Pull request merged**
